### DISCUSSION TITLE
Updating the outliner

### DISCUSSION THREAD

**Discussion by cking-spire** [Autodesk/maya-usd#2153] [Q&A] [answered]
Hi, 
I am attempting to write a python script that finds a USD file in our custom asset management system and then add a new prim to the stage and adds a reference to that prim pointing at the file. 
I have all this working by making calls to the USD APIs. However, the outliner wont refresh until I collapse a item and open it again. Is there a way for me to ask for the outliner to be refreshed?

Thanks,
Chris King
Spire Animation Studios

**Comment by neilrubie** [comment, marked as answer]
Give this function a try:

```
import maya.cmds as cmds

def refresh_outliners():
    """ Refresh all outliners in the session """
    editors = cmds.lsUI(editors=True)
    for editor in editors:
        if cmds.outlinerEditor(editor, exists=True):
            cmds.outlinerEditor(editor, edit=True, refresh=True)
```

**Comment by santosg87** [comment]
@CKing-Spire what version of the plugin/Maya are you in? the worakround above will do the job, but ideally, this should be working without the need of a workaround. would you be able to send me a sample of your workflow or a simplified version of the scripts you have that aren't updating the outliner, possibly? i would like to investigate this a bit further. :)

**Reply by cking-spire** [reply]
Hi George, 
I am using Maya 2022.1 and the mayaUsd that ships with that. I think this is mayaUsd 0.10.0.

My code look something like this:
        stage = mayaUsd.ufe.getStage(proxyUsdShapePath)
        newPrim = stage.DefinePrim(primPath)
        newPrim.GetReferences().AddReference(filePath)
        newPrim.SetInstanceable(True)
        refresh_outliners()

Thanks,
Chris
